cli/tests/test_admin_keygen.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import argparse
+import os
+import unittest
+
+from sawtooth_signing import secp256k1_signer as signing
+from sawtooth_cli.admin_command import keygen
+from sawtooth_cli.admin_command.utils import ensure_directory
+from sawtooth_cli.exceptions import CliException
+
+
+class TestKeygen(unittest.TestCase):
+    def __init__(self, test_name):
+        super().__init__(test_name)
+        self._temp_dir = None
+        self._parser = None
+        self._key_dir = ensure_directory('etc/keys', '/etc/sawtooth/keys')
+        self._key_name = 'test_key'
+        self._wif_filename = os.path.join(self._key_dir,
+                                          self._key_name + '.wif')
+        self._addr_filename = os.path.join(self._key_dir,
+                                           self._key_name + '.addr')
+
+    def setUp(self):
+        # self._temp_dir = tempfile.mkdtemp()
+
+        self._parser = argparse.ArgumentParser(add_help=False)
+
+        parent_parser = argparse.ArgumentParser(prog='test_keygen',
+                                                add_help=False)
+
+        subparsers = self._parser.add_subparsers(title='subcommands',
+                                                 dest='command')
+
+        keygen.add_keygen_parser(subparsers, parent_parser)
+
+    def tearDown(self):
+        # shutil.rmtree(self._temp_dir)
```

Comment:
If you are not using the tmp dir, be sure to clean up any files created (putting them in the tmp dir made this easier)